compiler/rustc_middle/src/ty/closure.rs

```diff
@@ -155,6 +155,54 @@ impl CapturedPlace<'tcx> {
         place_to_string_for_capture(tcx, &self.place)
     }
 
+    /// Returns mangled names of captured upvars. Here are some examples:
+    ///  - `_captured_val__name__field`
+    ///  - `_captured_ref__name__field`
+    ///
+    /// The purpose is to use those names in debuginfo. They should be human-understandable.
+    /// Without the names, the end users may get confused when the debuggers just print some
+    /// pointers in closures or generators.
+    pub fn to_mangled_name(&self, tcx: TyCtxt<'tcx>) -> String {
+        let prefix = match self.info.capture_kind {
+            ty::UpvarCapture::ByValue(_) => "_captured_val__",
+            ty::UpvarCapture::ByRef(_) => "_captured_ref__",
```

Comment:
Ok, to bikeshed a bit more.. I don't think `_captured_val__` is _as_ useful as `_captured_ref__` (especially in the common case of no field projections), and I'm also wondering if we can shorten the latter to just `_ref__`. I don't feel too strongly about it though.

Reply:
At least it looks more consistent (in my opinion)? But anyway, I'll remove `_captured_val__` and shorten `_captured_ref__` to `_ref__` if you like it.